commons-core/jvm/src/main/scala/com/avsystem/commons/concurrent/ObservableBlockingIterator.scala

```diff
@@ -0,0 +1,98 @@
+package com.avsystem.commons
+package concurrent
+
+import java.util.concurrent.ArrayBlockingQueue
+
+import com.avsystem.commons.collection.CloseableIterator
+import com.github.ghik.silencer.silent
+import monix.execution.{Ack, Scheduler}
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.blocking
+import scala.concurrent.duration.TimeUnit
+
+/**
+  * An `Iterator` backed by a `BlockingQueue` backed by an `Observable`.
+  * This essentially turns an `Observable` into an `Iterator`
+  * (which requires blocking so use this only as a last resort).
+  */
+class ObservableBlockingIterator[T](
```

Comment:
I've seen an alternative implementation in RxJava `https://github.com/ReactiveX/RxJava/blob/3.x/src/main/java/io/reactivex/rxjava3/internal/operators/observable/BlockingObservableIterable.java`. You might want to take a look or borrow some tests for better coverage :eyes:

Reply:
Quite a different implementation though because this one isn't based on Monix